sims/verilator/Makefile

```diff
@@ -1,112 +1,175 @@
+# Extra make variables:
+#
+# ENABLE_PRINTF_PATTERN - python regex. any printf() with text that matches
+#                         this regex will be force enabled in sim output
+#                         default is no selectively enabled printf's
+# VL8_PROFILE           - 'none' if no verilator profiling (default)
+#                         'all' if full verilator runtime profiling
+#                         'threads' if runtime thread profiling only
+# VL8_THREADS           - how many threads the simulator will use (default 1)
+# VL8_FST_MODE          - enable FST waveform instead of VCD
+#============================================================================
+
+
+#============================================================================
 # verilator makefile
-#########################################################################################
+#============================================================================
 ifeq ($(shell which verilator),)
-$(error Did not find Verilator in PATH. Make sure all requirements are installed)
+	$(error Did not find Verilator in PATH. Make sure all requirements are installed)
 endif
 
-#########################################################################################
+#============================================================================
 # general path variables
-#########################################################################################
+#============================================================================
 base_dir=$(abspath ../..)
 sim_dir=$(abspath .)
 
-#########################################################################################
+#============================================================================
 # include shared variables
-#########################################################################################
+#============================================================================
 include $(base_dir)/variables.mk
 
-#########################################################################################
+#============================================================================
 # name of simulator (used to generate *.f arguments file)
-#########################################################################################
+#============================================================================
 sim_name = verilator
 
-#########################################################################################
-# vcs simulator types and rules
-#########################################################################################
+#============================================================================
+# verilator simulator types and rules
+#============================================================================
 sim_prefix = simulator
 sim = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)
 sim_debug = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)-debug
 
-PERMISSIVE_ON=
```

Comment:
I believe this will break running simulations.

Reply:
how will it break? verilator doesn't use PERMISSIVE

Reply:
VCS uses the `PERMISSIVE` flags. I don't think it will break anything (lets see what the CI says) since make expands it to an empty string. However, I do like it defined and kept so that all make recipes can use the same set of variables and so that future users can see where the variables originate from. For example, in `common.mk` we use `run-binary` with `PERMISSIVE_ON/OFF`. Future users might see the variables and wonder where they originate from. I think this is more of a clarity issue.

Reply:
the opposing argument is that it advertises a feature in verilator that does not exist, which seems a bit cargo-cultish. 

removing PERMISSIVE from verilator will help future developers figure out more quickly why permissive exists in the first place

Reply:
TIL what `cargo-cultish` means.

I'm flexible on removing this as long as things don't break.

Reply:
I'm actually inclined to agree it should be deleted (perhaps even renamed to `VCS_PERMISSIVE`)

Reply:
Is that not just because verilator doesn't currently take any runtime arguments?
Can you pass it one of the verilator runtime flags like: `+verilator+rand+reset+0` just for my edification?

Reply:
i'm leaving it deleted an opening a github issue on this

Reply:
it looks like to add custom plusargs to verilator you need to use https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/util/PlusArg.scala in the chisel code.

Reply:
the https://www.veripool.org/projects/verilator/wiki/Manual-verilator#SIMULATION-RUNTIME-ARGUMENTS probably make the simulator crash since they are not whitelisted plusargs.

Reply:
Yeah we have a related `testchipip` issue for that https://github.com/ucb-bar/testchipip/issues/86 . We need to bump rocket-chip to get most of the usable features in PlusArg.scala

Reply:
So we do need permissive and its even slightly different for verilator so its good that its overrideable:
```
simulator-example-RocketConfig +max-cycles=5000 +verbose -- +permissive +verilator+V +permissive-off output/example.TestHarness.RocketConfig/rv64ui-p-addi
using random seed 1590055841
  Version: Verilator 4.016 2019-06-16
%Error: COMMAND_LINE:0: Exiting due to command line argument (not an error)
Aborting...
Aborted (core dumped)
```
versus
```
simulator-example-RocketConfig +max-cycles=5000 +verbose +verilator+V output/example.TestHarness.RocketConfig/rv64ui-p-addi
simulator-example-RocketConfig: invalid plus-arg (Verilog or HTIF) "+verilator+V"
```

Reply:
lol why did we set it to empty? @jerryz123 
https://github.com/ucb-bar/chipyard/commit/340ed90652c58ec3228706f49463cb12920f1307

Reply:
i've added these back since `common.mk` expects it to be set. i missed that.